internal/provider/resource_gitlab_tag.go

```diff
@@ -0,0 +1,123 @@
+package provider
+
+import (
+	"context"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	gitlab "github.com/xanzy/go-gitlab"
+)
+
+var _ = registerResource("gitlab_tag", func() *schema.Resource {
+	return &schema.Resource{
+		Description: "This resource allows you to create and manage GitLab tags.",
```

Comment:
```suggestion
		Description: `This resource allows you to create and manage GitLab tags.
		
**Upstream API** : [GitLab API docs](https://docs.gitlab.com/ee/api/tags.html)`,
```

Reply:
Are you sure that you name resources with the pattern `gitlab_project_`*? In the previous pull request resource was named `gitlab_brunch`.

Reply:
Yeah, I've also noticed. Might be okay to go with tags. Anyways, as I've said, it's not a showstopper for me.

Reply:
I've applied this. Can you applied it too for the data source and rerun `make reviewable`?